src/alacris_lib/state_update.ml

```diff
@@ -1,99 +1,120 @@
+let rec inner_state_update_periodic_loop : Address.t -> unit Lwt_exn.t =
+  fun operator ->
+  let open Lwt_exn in
+  retrieve_validated_rev_digest ()
+  >>= fun (operator_revision, operator_digest) -> post_state_update ~operator ~operator_revision ~operator_digest
+  >>= fun () -> sleep_delay_exn Side_chain_server_config.state_update_period_in_seconds_f
+  >>= fun () -> inner_state_update_periodic_loop operator
+
+let rec inner_state_update_nocheck_periodic_loop : Address.t -> unit Lwt_exn.t =
+  fun operator ->
   let open Lwt_exn in
-  get_contract_address ()
-  >>= fun contract_address ->
-  let operation = make_state_update_call ~contract_address ~operator_digest ~operator_revision in
-  let oper_addr = Side_chain_server_config.operator_address in
-  Ethereum_user.post_operation ~operation ~sender:oper_addr ~value:TokenAmount.zero
-
-
-let inner_state_update_request_loop () =
-  let open Lwt in
-  let digest_entry_ref : digest_entry ref = ref {revision=Revision.zero; digest=Digest.zero} in
-  let rec inner_loop : unit -> unit Lwt.t =
-    fun () ->
-    Lwt_mvar.take request_state_update_mailbox
-    >>= function
-    | GetLastRevision (rev_u : Revision.t Lwt.u) ->
-       Lwt.wakeup_later rev_u !digest_entry_ref.revision;
-       inner_loop ()
-    | GetLastCommit (digest_u : Digest.t Lwt.u) ->
-       Lwt.wakeup_later digest_u !digest_entry_ref.digest;
-       inner_loop ()
-    | Submit ((new_digest, notify_u) : (Digest.t * TransactionReceipt.t OrExn.t Lwt.u)) ->
-       let new_rev = Revision.add !digest_entry_ref.revision Revision.one in
-       let new_digest_entry = {revision=new_rev; digest=new_digest} in
-       digest_entry_ref := new_digest_entry;
-       let revision : Revision.t = Revision.of_int 742 in
-       (* TODO: Clarify this. The value 742 is here for fun so that we know what problem happen.
-        In reality, we need the state revision *)
-       post_state_update revision new_digest
-       >>= fun ereceipt ->
-       Lwt.wakeup_later notify_u ereceipt;
-       inner_loop ()
-  in inner_loop ()
-
-
-let start_state_update_daemon () =
+  retrieve_validated_rev_digest ()
+  >>= fun (operator_revision, operator_digest) -> post_state_update_nocheck ~operator ~operator_revision ~operator_digest
+  >>= fun () -> sleep_delay_exn Side_chain_server_config.state_update_period_in_seconds_f
+  >>= fun () -> inner_state_update_nocheck_periodic_loop operator
+
+let start_state_update_periodic_daemon address =
+  register_keypair "alice" Signing.Test.alice_keys;
+  register_keypair "bob" Signing.Test.bob_keys;
   if state_update_log then
-    Logging.log "Beginning of start_state_update_daemon";
-  Lwt.async inner_state_update_request_loop;
+    Logging.log "Beginning of start_state_update_periodic_operator wait=%f" Side_chain_server_config.state_update_period_in_seconds_f;
+  Lwt.async (fun () -> inner_state_update_periodic_loop address);
   Lwt_exn.return ()
 
+let start_state_update_nocheck_periodic_operator address =
```

Comment:
I renamed as **_for_test**.
The problem is the following: The tests run with their own state update but they use the same ethereum blockchain for making their tests.
However, with the use of require statement in the contract we cannot have a state update done two times. The first state update is always the same since we start from the same null state. Thus for the test we should not run it at all.
Thus for the tests, we need the state update to be done differently for the normal run and the main tests. This is why we need a separate function.